packages/eventual-send/src/index.d.ts

```diff
@@ -18,6 +18,20 @@ export type EOnly<T> = T extends (...args: infer P) => infer R
     >
   : ERef<T>;
 
+type FilteredKeys<T, U> = {
+  [P in keyof T]: T[P] extends U ? P : never;
+}[keyof T];
+
+type DataOnly<T> = Omit<T, FilteredKeys<T, Function>>;
```

Comment:
I'm always wary of using `Function` in these cases. Why not use `(...args: any[]) => any` ?

Reply:
> wary

Why? What risk would it prevent in this case?

> Why not use `(...args: any[]) => any` ?
`Function` reads much more clearly and there's no benefit to the longer form.

Reply:
My understanding is that `Function` is anything that has the shape of the `Function` prototype, aka has `.call`, `.apply`, `.bind` etc. `(...args: any[]) => any` is anything that is callable.

This is similar to an object with properties being different than something that extends `Object`.

See this [playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBCBGArGBeGBvAUDHMwEMBbAUwC4YAiBRCgGm1wBtiwBzKAC3IAZ7cZ8AB0GMAngAoA+pwCWEcvjCiAlBgC+fXMHyNGU2fIFLV6DQxzwZYACb6Oc0opUZz-GACdiUAK7uwcJABuVzN+KBAAZSh3K1ZxE1d+Tx8-SgAxfABrYhg07zBgKBlwCmD+UNwAbQjRQngQRgA6DnwIAElIKEVgYgBdeJc3XGTffwAzHQhiMtwKnEF3EHCoUUEyXPzC4rBGhaWl1eJNHHx3Vm8SMChDSt7jmG1dYndyAZQAPnV6NUxMFbWAog2hA8gUiuBUDB-sQQGNATBiAAPKAsawQDZg7YwAD8UPc3hy5AmjCmwUwoE6MDG+XIoK2ELQ1ECQA).

Reply:
Right. In this case I think that's what we want. We're partitioning things that are callable from those that aren't.